### PR TITLE
Process files using extensions and targetExtension options

### PR DESCRIPTION
**Pull request by marcioj**
//cc @stefanpenner

### CODE REVIEW

**Comment by stefanpenner** [comment]
filter has .extensions for this: https://github.com/broccolijs/broccoli-filter/blob/master/index.js#L18-L19

File: index.js [hunk 1]
```diff
@@ -181,7 +181,7 @@ module.exports = CachingWriter.extend({
 
     walkSync(inDir)
       .forEach(function(relativePath) {
-        if (relativePath.slice(-1) !== '/') {
+        if (relativePath.slice(-1) !== '/' && relativePath.slice(-3) === '.js') {
```
Comment (rwjblue) [inline index.js]: I believe this can use only the .js condition (it is not possible to end with '/' **and** '.js').

Comment (marcioj) [inline index.js]: Shame myself :see_no_evil:

**Comment by marcioj** [comment]
@stefanpenner updated to use `extensions` and `targetExtension` options.
Let me know if there is something to improve here

**Comment by marcioj** [comment]
Thanks @stefanpenner @rwjblue !

**Comment by stefanpenner** [comment]
released as `v0.6.0`

**Pull request merged**